packages/data/src/registry.js

```diff
@@ -74,18 +74,19 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 	/**
 	 * Calls a selector given the current state and extra arguments.
 	 *
-	 * @param {string} reducerKey Part of the state shape to register the
-	 *                            selectors for.
+	 * @param {string|Object} storeName Unique namespace identifier for the store
+	 *                                  or the store definition.
 	 *
 	 * @return {*} The selector's returned value.
 	 */
-	function select( reducerKey ) {
-		const store = stores[ reducerKey ];
+	function select( storeName ) {
+		storeName = String( storeName );
```

Comment:
An alternative would be along those lines
```suggestion
		const storeName = isObject( storeNameOrDefinition ) ? storeNameOrDefinition.name : storeNameOrDefinition;
```